core/Segment.php

```diff
@@ -297,6 +342,23 @@ protected function getCleanedExpression($expression)
         $segment = $this->getSegmentByName($name);
         $sqlName = $segment['sqlSegment'];
 
+        // Build subqueries for segments that are not on log_visit table but use !@ or != as operator
+        // This is required to ensure segments like actionUrl!@value really do not include any visit having an action containing `value`
+        if ($this->doesSegmentNeedSubquery($matchType, $name)) {
```

Comment:
btw do you reckon it's possible to group all segments together to have only one `NOT IN` subquery? Not too important if too difficult. If not too difficult that could be a good improvement when using several `NOT IN` and `NOT EQUALS` . Might get difficult though with changing operands etc but maybe the ones that are connected with OR.

eg 
![image](https://user-images.githubusercontent.com/273120/86415992-d47e4380-bd1c-11ea-8944-896f303528f1.png)

I wonder if it was easily possible to not have 4 `NOT IN` queries but only 2 `NOT IN` queries. (implementing only one `NOT IN` would probably get very tricky).

Meaning as soon as we detect a `NOT IN` subquery, and more segments are following with `OR`, then these could be maybe grouped together? Maybe check if that's not too tricky and be good to have tests for it.

Reply:
Well having `action url does not contain` combined with an OR actually cancels each other out, but I got your point. Will have a look if that's easy to solve

Reply:
While trying to implement that I realized, that we actually can't combine two action segments in an subquery if they are combined with OR. We could combine them if they are using an AND. But guess that would actually require to parse the tree completely, as we can't combine them, if one of them is concatenated with another segment using OR 🤯

Reply:
>  But guess that would actually require to parse the tree completely, as we can't combine them, if one of them is concatenated with another segment using OR 🤯

Getting complicated then... was hoping we could combine at least the `OR` ones. All good for now if not doable